src/v/security/gssapi_authenticator.cc

```diff
@@ -0,0 +1,511 @@
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "security/gssapi_authenticator.h"
+
+#include "bytes/bytes.h"
+#include "config/configuration.h"
+#include "kafka/protocol/request_reader.h"
+#include "security/acl.h"
+#include "security/errc.h"
+#include "security/gssapi.h"
+#include "security/logger.h"
+#include "vlog.h"
+
+#include <boost/outcome/basic_outcome.hpp>
+#include <boost/outcome/success_failure.hpp>
+#include <fmt/ranges.h>
+#include <gssapi/gssapi.h>
+#include <gssapi/gssapi_ext.h>
+
+#include <array>
+#include <sstream>
+#include <string_view>
+
+namespace security {
+
+std::ostream&
+operator<<(std::ostream& os, gssapi_authenticator::state const s) {
+    using state = gssapi_authenticator::state;
+    switch (s) {
+    case state::init:
+        return os << "init";
+    case state::more:
+        return os << "more";
+    case state::ssfcap:
+        return os << "ssfcap";
+    case state::ssfreq:
+        return os << "ssfreq";
+    case state::complete:
+        return os << "complete";
+    case state::failed:
+        return os << "failed";
+    }
+}
+
+static void display_status_1(std::string_view m, OM_uint32 code, int type) {
+    while (true) {
+        OM_uint32 msg_ctx{0};
+        OM_uint32 min_stat{};
+        gss::buffer msg;
+        auto maj_stat = gss_display_status(
+          &min_stat, code, type, GSS_C_NO_OID, &msg_ctx, &msg);
+        if (maj_stat != GSS_S_COMPLETE) {
+            vlog(seclog.info, "gss status from {}", m);
+            break;
+        } else {
+            vlog(seclog.info, "GSS_API error {}: {}", m, msg);
+        }
+
+        if (!msg_ctx) {
+            break;
+        }
+    }
+}
+
+std::string display_ctx_flags(OM_uint32 flags) {
+    std::stringstream ss;
+    if (flags & GSS_C_DELEG_FLAG) ss << "context flag: GSS_C_DELEG_FLAG\n";
+    if (flags & GSS_C_MUTUAL_FLAG) ss << "context flag: GSS_C_MUTUAL_FLAG\n";
+    if (flags & GSS_C_REPLAY_FLAG) ss << "context flag: GSS_C_REPLAY_FLAG\n";
+    if (flags & GSS_C_SEQUENCE_FLAG)
+        ss << "context flag: GSS_C_SEQUENCE_FLAG\n";
+    if (flags & GSS_C_CONF_FLAG) ss << "context flag: GSS_C_CONF_FLAG \n";
+    if (flags & GSS_C_INTEG_FLAG) ss << "context flag: GSS_C_INTEG_FLAG \n";
+    return ss.str();
+}
+
+struct hex {
+    explicit hex(bytes const& v)
+      : _v{v} {}
+    explicit hex(bytes_view const& v)
+      : _v{v} {}
+    friend std::ostream& operator<<(std::ostream& os, hex const& h) {
+        for (size_t i = 0; i < h._v.length(); ++i) {
+            if ((i % 16) == 0) {
+                os << '\n';
+            }
+            os << std::setfill('0') << std::setw(2) << std::hex
+               << (int)(h._v[i]) << ' ';
+        }
+        return os;
+    }
+    bytes_view _v;
+};
```

Comment:
i was thinking we had some hex utilities already. should this go with those (if they exist?)

Reply:
Thanks. Should probably one-line this if I keep it.

Reply:
I've removed this and logged the length instead.